pkg/accounts/adaptor/repository/prisma.ts

```diff
@@ -0,0 +1,221 @@
+      where: {
+        name: name,
+      },
+    });
+    if (!res) {
+      return Option.none();
+    }
+    return Option.some(this.fromPrismaArgs(res));
+  }
+
+  private toPrismaArgs(account: Account): AccountPrismaArgs {
+    const role = (
+      {
+        normal: 0,
+        moderator: 1,
+        admin: 2,
+      } satisfies Record<AccountRole, number>
+    )[account.getRole];
+
+    const status = (
+      {
+        active: 0,
+        notActivated: 1,
+      } satisfies Record<AccountStatus, number>
+    )[account.getStatus];
+
+    const frozen = (
+      {
+        normal: 0,
+        frozen: 1,
+      } satisfies Record<AccountFrozen, number>
+    )[account.getFrozen];
+
+    const silenced = (
+      {
+        normal: 0,
+        silenced: 1,
+      } satisfies Record<AccountSilenced, number>
+    )[account.getSilenced];
+
+    return {
+      id: account.getID,
+      name: account.getName,
+      nickname: account.getNickname,
+      mail: account.getMail,
+      passphrase_hash: account.getPassphraseHash ?? '',
+      bio: account.getBio,
+      role: role,
+      frozen: frozen,
+      silenced: silenced,
+      status: status,
+      created_at: account.getCreatedAt,
+      updated_at: !account.getUpdatedAt ? null : account.getUpdatedAt,
+      deleted_at: !account.getDeletedAt ? null : account.getDeletedAt,
+    };
+  }
+
+  private fromPrismaArgs(args: AccountPrismaArgs): Account {
+    const role =
+      (
+        {
+          0: 'normal',
+          1: 'moderator',
+          2: 'admin',
+        } satisfies Record<number, AccountRole>
+      )[args.role] ?? 'normal';
+
+    const status =
+      (
+        {
+          0: 'active',
+          1: 'notActivated',
+        } satisfies Record<number, AccountStatus>
+      )[args.status] ?? 'active';
+
+    const frozen =
+      (
+        {
+          0: 'normal',
+          1: 'frozen',
+        } satisfies Record<number, AccountFrozen>
+      )[args.frozen] ?? 'normal';
+
+    const silenced =
+      (
+        {
+          1: 'silenced',
+          0: 'normal',
+        } satisfies Record<number, AccountSilenced>
+      )[args.silenced] ?? 'normal';
+
+    return Account.reconstruct({
+      id: args.id as ID<AccountID>,
+      name: args.name as AccountName,
+      nickname: args.nickname,
+      mail: args.mail,
+      passphraseHash:
+        args.passphrase_hash === null ? undefined : args.passphrase_hash,
+      bio: args.bio,
+      role: role,
+      frozen: frozen,
+      silenced: silenced,
+      status: status,
+      createdAt: args.created_at,
+      deletedAt: args.deleted_at === null ? undefined : args.deleted_at,
+      updatedAt: args.updated_at === null ? undefined : args.updated_at,
+    });
+  }
+}
+
+export class PrismaAccountVerifyTokenRepository
+  implements AccountVerifyTokenRepository
+{
+  private readonly prisma: PrismaClient;
+
+  constructor(prisma: PrismaClient) {
+    this.prisma = prisma;
+  }
```

Comment:
```suggestion
  constructor(private readonly prisma: PrismaClient) {}
```
こちらもコンストラクタ引数を直接メンバーにしておきましょう